src/instructlab/training/main_ds.py

```diff
@@ -12,8 +12,16 @@
 
 # Third Party
 from accelerate import Accelerator
-from deepspeed.ops.adam import DeepSpeedCPUAdam, FusedAdam
-from deepspeed.runtime.zero.utils import ZeRORuntimeException
+try:
+    from deepspeed.ops.adam import DeepSpeedCPUAdam, FusedAdam
+    from deepspeed.runtime.zero.utils import ZeRORuntimeException
+except ImportError:
+    DeepSpeedCPUAdam = None
+    FusedAdam = None
+    ZeRORuntimeException = None
+    local_rank = int(os.getenv('LOCAL_RANK', -1))
```

Comment:
`LOCAL_RANK` is typically populated by `torchrun`, so you'll want to handle the case where it may not be set - in which case we should just assume that we are the main process:
```suggestion
    local_rank = int(os.getenv('LOCAL_RANK', None))
    if __name__ == '__main__' and (not local_rank or local_rank == 0):
```